kafka-scheduler/src/main/java/com/mesosphere/dcos/kafka/offer/PersistentOfferRequirementProvider.java

```diff
@@ -20,475 +19,462 @@
+        final Map<String, String> newEnvMap = OfferUtils.fromEnvironmentToMap(oldEnvironment);
+        newEnvMap.putAll(getUpdatedTaskEnvironment(config, taskBuilder.getCommand().getEnvironment()));
 
-    // Copy everything except config target label
-    for (Label label : taskBuilder.getLabels().getLabelsList()) {
-      String key = label.getKey();
-      String value = label.getValue();
+        CommandInfo.Builder cmdBuilder = CommandInfo.newBuilder(existingCommandInfo);
 
-      if (!key.equals(CONFIG_TARGET_KEY)) {
-        labelMap.put(key, value);
-      }
+        String brokerCmd = getBrokerCmd(config);
+        cmdBuilder.setValue(brokerCmd);
+
+        cmdBuilder.setEnvironment(OfferUtils.environment(newEnvMap));
+        taskBuilder.setCommand(cmdBuilder.build());
+        taskBuilder.clearData();
+        return taskBuilder;
     }
 
-    labelMap.put(CONFIG_TARGET_KEY, configName);
-    taskBuilder.setLabels(labels(labelMap));
-    return taskBuilder;
-  }
+    private ExecutorInfo.Builder updateExecutorCmd(
+            ExecutorInfo.Builder updatedExecutor,
+            KafkaSchedulerConfiguration config,
+            String configName) throws ConfigStoreException {
+
+        final CommandInfo existingCommandInfo = updatedExecutor.getCommand();
+        final Environment oldEnvironment = existingCommandInfo.getEnvironment();
+        final Map<String, String> newEnvMap = OfferUtils.fromEnvironmentToMap(oldEnvironment);
+        newEnvMap.putAll(getUpdatedExecutorEnvironment(config, configName));
 
-  private TaskInfo.Builder updateCmd(TaskInfo.Builder taskBuilder, String configName)
-          throws InvalidRequirementException {
-    Map<String, String> envMap = new HashMap<>();
+        CommandInfo.Builder cmdBuilder = CommandInfo.newBuilder(existingCommandInfo);
+        cmdBuilder.setEnvironment(OfferUtils.environment(newEnvMap))
+                .clearUris()
+                .addAllUris(getUpdatedUris(config));
 
-    try {
-      final CommandInfo existingCommandInfo = CommandInfo.parseFrom(taskBuilder.getData());
-      for (Variable variable : existingCommandInfo.getEnvironment().getVariablesList()) {
-        if (variable.getName().equals(CONFIG_ID_KEY)) {
-          envMap.put(CONFIG_ID_KEY, configName);
+        return updatedExecutor.setCommand(cmdBuilder);
+    }
+
+    private static Long getDynamicPort() {
+        return 9092 + ThreadLocalRandom.current().nextLong(0, 1000);
+    }
+
+    private TaskInfo getNewTaskInfo(KafkaSchedulerConfiguration config, String configName, int brokerId)
+            throws IOException, URISyntaxException {
+
+        BrokerConfiguration brokerConfiguration = config.getBrokerConfiguration();
+        String brokerName = OfferUtils.brokerIdToTaskName(brokerId);
+        String role = config.getServiceConfiguration().getRole();
+        String principal = config.getServiceConfiguration().getPrincipal();
+
+        String containerPath = "kafka-volume-" + UUID.randomUUID();
```

Comment:
Nit-pick: Maybe make `"kafka-volume"` a constant ? I see that it is also being used in a test.